src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -568,10 +617,25 @@ private void uploadFleetStatusServiceData(Set<GreengrassService> greengrassServi
                 .messageType(MessageType.fromTrigger(trigger))
                 .deploymentInformation(deploymentInformation)
                 .build();
+        // make sure messages are not published too frequently to cause message drop from cloud
+        if (lastFSSPublishTime.plusSeconds(FLEET_STATUS_MESSAGE_PUBLISH_MAX_WAIT_TIME_SEC)
+                .isBefore(Instant.now())) {
+            publisher.publish(fleetStatusDetails, components);
+            lastFSSPublishTime = Instant.now();
+            logger.atInfo().event("fss-status-update-published").kv("trigger", trigger)
+                    .log("Status update published to FSS");
+        } else {
+            // Calculate the time that next message needs to wait.
+            Duration duration = Duration.between(Instant.now(), lastFSSPublishTime
```

Comment:
I dont understand how this prevents messages from being dropped by FSS. This is simply scheduling multiple messages to be published at the same time, right?